### ISSUE TITLE
Non-split thumbnail view in FileSystem M.I.A.

### ISSUE THREAD

**Issue by somnivore** [godotengine/godot#23627]
**Godot version:**

a16d9c6ab68cadb3cff07ef8b7b8c00717e98922

**OS/device including version:**

Linux

**Issue description:**

In previous versions of Godot, it was possible to double-click a folder in tree-view to enter thumbnail-view:

![image](https://user-images.githubusercontent.com/28724859/48298085-813aeb80-e47c-11e8-90ba-d0f1d0ab953f.png)
![image](https://user-images.githubusercontent.com/28724859/48298089-96b01580-e47c-11e8-8aa7-46cfa19b0b52.png)

Currently, double-clicking a folder in tree-view, instead, reveals or hides the contents of that folder in tree-view.  Right-clicking a folder and selecting 'Open' in the dialogue also has this behavior.  The only way to access thumbnail-view anymore is to put FileSystem into split mode:

![image](https://user-images.githubusercontent.com/28724859/48298097-dc6cde00-e47c-11e8-8817-831bb8c1593e.png)

In other words, you can have tree-view only, or tree-view and thumbnail-view, but not thumbnail-view only; there seems to be no way to access this mode anymore.

![image](https://user-images.githubusercontent.com/28724859/48298164-efcc7900-e47d-11e8-976b-3edc998fa07f.png)

**Comment by groud** [comment]
This was removed on purpose. If too many people miss it, we might reconsider making it available via a button or something like that.

**Comment by somnivore** [comment]
I certainly would prefer it.  Though to be frank, if I could put the FileSystem in the same area as Output and Debugger, I'd be even happier.

From what I remember, we used to hide the tree-view in split-mode if the FileSystem was small enough.  This is because it doesn't make a lot of sense to keep both open at the same time when space is limited, it becomes too cramped:

![image](https://user-images.githubusercontent.com/28724859/48318885-2e704980-e5cc-11e8-8269-bb377653a8e0.png)

I could do the following, where the tree is just a sliver, but it looks a bit silly, especially when the previous behavior did the job just fine:
![image](https://user-images.githubusercontent.com/28724859/48318893-4cd64500-e5cc-11e8-91da-5a20ad2cac0f.png)

**Comment by Calinou** [comment]
Feature and improvement proposals for the Godot Engine are now being discussed and reviewed in a dedicated [**Godot Improvement Proposals (GIP)**](https://github.com/godotengine/godot-proposals) ([godotengine/godot-proposals](https://github.com/godotengine/godot-proposals)) issue tracker. The GIP tracker has a detailed issue template designed so that proposals include all the relevant information to start a productive discussion and help the community assess the validity of the proposal for the engine.

The main ([godotengine/godot](https://github.com/godotengine/godot)) tracker is now solely dedicated to bug reports and Pull Requests, enabling contributors to have a better focus on bug fixing work. Therefore, we are now closing all older feature proposals on the main issue tracker.

If you are interested in this feature proposal, please [**open a new proposal on the GIP tracker**](https://github.com/godotengine/godot-proposals/issues/new/choose) following the given issue template (after checking that it doesn't exist already). Be sure to reference this closed issue if it includes any relevant discussion (which you are also encouraged to summarize in the new proposal). Thanks in advance!
